### ISSUE TITLE
Add circuit breaking error handling to ReactiveSeq

### ISSUE THREAD

**Issue by johnmcclean** [aol/cyclops#988]
Dedicated operators to switch on error and handle errors

```java 
* onErrorSwitch(Function<? super Throwable,? extends Publisher<T>> fn)
* onErrorSwitch(Publisher<T> pub)
```

```java 
* onError(Consumer<? super Throwable> consumer)
```

**Comment by johnmcclean** [comment]
```java 
* onErrorSwitch(Publisher<T> pub)
```

Should be called

```java 
* recoverWith(Supplier<Publisher<T>> pub)
```
For consistency with other Operators in cyclops.

An onErrorBreak operator is required (at least internally to force circuit breaking behaviour).

**Comment by johnmcclean** [comment]
Released in 10.3.0
